File: LongestCommonPrefix.java
public class test {
    public static void main(String[] args) {
        String[] str = {"flower", "flight", "flow"};
        char[] charArray = new char[str.length];
        String extractedStr = "" , newStr = "", minimumLengthWord = str[0];
        for(int i = 1; i < str.length; i++) {
            if(minimumLengthWord.length() > str[i].length()) {
                minimumLengthWord = str[i];
            }
        }
        for(int i =0; i< minimumLengthWord.length(); i++) {
            int flag = 0;
            for(int j = 0; j < str.length; j++) {
                extractedStr = str[j];
                charArray[j] = extractedStr.charAt(i);
            }
            for(int k = 1; k < charArray.length; k++) {
                char character = charArray[0];
                if (character != charArray[k]) {
                    flag = 1;
                }
            }
            if (flag == 0) {
                newStr += charArray[0];
            }
        }
        System.out.println(newStr);
    }
}

File: LongestCommonPrefix.py
str = ["flower", "flight", "flow"]
charArray = [0] * len(str)
newString = ""

for i in range (len(str)):
    minimumLengthWord = str[0]
    if minimumLengthWord > str[i]:
        minimumLengthWord = str[i]
for i in range(len(minimumLengthWord)):
    flag = 0
    for j in range(len(str)):
        extractedString = str[j]
        charArray[j] = extractedString[i]
    for k in range(1, len(charArray)):
        character = charArray[0]
        if character != charArray[k]:
            flag = 1
    if flag == 0:
        newString += character

print(newString)
